go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -124,7 +124,7 @@ func (bsnc *BaseSecondaryNetworkController) ensurePodForSecondaryNetwork(pod *ka
 		return nil
 	}
 
-	if bsnc.lsManager.IsNonHostSubnetSwitch(switchName) {
+	if bsnc.doesNetworkRequireIPAM() && bsnc.lsManager.IsNonHostSubnetSwitch(switchName) {
```

Comment:
This probably needs a similar comment as above.

Reply:
I think that we can't tell apart if it is a NonHostSubnet or if it is a IPAM less network and I am not sure if this means we need a separate flag to tell them apart.

Reply:
I am not 100% sure so I need to ask.

But it might mean be that we don't need to process the attachment of pods on those hosts. Like if those hosts were handled by a different CNI. 

In this case we would really need to tell the nonHostSubnet case apart from the IPAM-less case.

Reply:
I think this confirms it: https://github.com/openshift/ovn-kubernetes/blob/a10433c51f38c2b73ea4019e69512d427d88542d/docs/hybrid-overlay.md#introduction

Reply:
After having read this part of the code, I'm unsure that is needed, since `noHostSubnet` is only relevant for the default network and secondary **layer3** networks - were we will **not** have IPAMless scenarios.

Like, all calls of [AddNoHostSubnetSwitch](https://github.com/ovn-org/ovn-kubernetes/blob/0680beed9290bca8a166e83cbb16d60f938281ec/go-controller/pkg/ovn/logical_switch_manager/logical_switch_manager.go#L130) comes from the `DefaultNetworkController` and from `SecondaryLayer3NetworkController` .

Here it maybe more readable if I compare the topology, instead of the IPAM presence.

Reply:
yeah noHostSubnets is for other kube nodes which we are not providing or allocating subnets for with OVN. Examples include windows nodes, or openshift-sdn nodes as we migrate from openshift-sdn to OVNK

Reply:
Agreed